src/main/java/com/pszymczyk/TripService.java

```diff
@@ -1,40 +1,27 @@
 package com.pszymczyk;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.UUID;
-import java.util.stream.Collectors;
+import java.util.Optional;
 
-public class TripService {
+import org.springframework.stereotype.Component;
 
-    private final TripDao tripDao;
+@Component
+class TripService {
 
-    public TripService(TripDao tripDao) {
-        this.tripDao = tripDao;
+    private final TripRepository tripRepository;
+
+    TripService(TripRepository tripRepository) {
+        this.tripRepository = tripRepository;
     }
 
-    public void book(String userId, String tripCode) {
-        Trip trip = tripDao.findTrip(tripCode);
-        List<Reservation> reservations = trip.getReservations()
-                                             .stream()
-                                             .filter(r -> r.getStatus() == Reservation.ReservationStatus.CONFIRMED)
-                                             .collect(Collectors.toList());
+    void book(String userId, String tripCode) {
+        Trip trip = tripRepository.findTrip(tripCode);
+
+        Optional<ReservationSummary> reservationSummary = trip.requestReservation(userId);
```

Comment:
Czy tripRepository.findTrip nie może zwrócić null?
Jeśli tak, to w 19 linii możemy mieć NullPointerException.

Reply:
Tak, może nawet dzisiaj omawialiśmy taki przypadek, good point.